### PR TITLE
Add Stone pieces to the game

### PR DESCRIPTION
**Pull request by ghyl**
Stone will spawn with at the top of the Board with a frequency dependent on the current level. Stones are removed via power-ups or automatically when they are at the bottom. More details are in the requirements document.

### CODE REVIEW

**Comment by albert-smit** [comment]
merge #68 first

**Comment by albert-smit** [comment]
still has a few pmd and findbugs violations and checkstyle warnings

**Comment by ghyl** [comment]
I solved most of the PMD errors, but some of them will probably be addressed in one of the other branches. Before the deadline I will make another checkstyle and pmd sweep, but I don't want to risk doing double work right now.

**Comment by albert-smit** [comment]
ok, I meant merging #68 into master

**Comment by albert-smit** [comment]
looks good, especially if the final pmd and checkstyle warnings are fixed in the final version, don't forget findbugs either.

File: src/main/java/nl/tudelft/ti2206/semgr8/model/piece/Stone.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * returns a negative color according to the position.
+     * This is done because otherwise a connected line of StoneGems will
+     * be removed.
+     *
+     * @return a int according to the position.
+     */
+    public int getColor() {
+        return -2;
```
Comment (wroel) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/piece/Stone.java]: if all stones have color -2, is it still impossible to get a connected line of stones?

Comment (ghyl) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/piece/Stone.java]: Yes, this is impossible. The `matchColor` method is used for matching colors, and I have made it return false when the game tries to match a stone to another piece.

**Comment by ghyl** [comment]
There seems to be some conflicts with master left, I'll resolve them now, but then it looks good to merge right?

**Comment by ghyl** [comment]
One travis is done, then this one can be merged. Everything is merged in it, but the levelprogressbar does not update after achieving the next level. It only updates after you have made a another swap at level 2.

**Comment by albert-smit** [comment]
I'll fix the update of the progressbar  
i'll merge it now

**Pull request merged**